### PR TITLE
Enforce java 11 due to openapi generator compatibility

### PR DESCRIPTION
**Pull request by maczikasz**
https://github.com/OpenAPITools/openapi-generator/pull/15553

https://github.com/gradle/ge/issues/27007

### CODE REVIEW

**Review by welandaz — APPROVED** [review]

**Review by c00ler — APPROVED** [review]

File: build.gradle.kts [hunk 1]
```diff
@@ -73,6 +82,16 @@ openApiGenerate {
     ))
 }
 
+tasks.named<Test>("test") {
```
Comment (c00ler) [inline build.gradle.kts:85]: Can be simplified: 

```suggestion
tasks.test {
```

File: src/test/java/com/gradle/enterprise/api/MockServer.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        ApiClient apiClient = new ApiClient();
+        apiClient.setServers(Collections.singletonList(new ServerConfiguration(
+            "http://" + remoteAddress.getHostName() + ":" + remoteAddress.getPort(), "mockServer", Collections.emptyMap()
+        )));
+        BuildsApi buildsApi = new BuildsApi(apiClient);
+        apiClient.addDefaultHeader("Authorization", "Bearer XYZ");
+
+        List<Build> builds = buildsApi.getBuilds(new BuildsQuery());
+
+        assert !builds.isEmpty();
```
Comment (c00ler) [inline src/test/java/com/gradle/enterprise/api/MockServer.java]: Why not use any assertion library? There is one in the `junit5` framework.

**Review by c00ler — COMMENTED** [review]
How exactly `Java 11` is enforced? It's not evident from the change.

File: src/test/java/com/gradle/enterprise/api/MockServer.java (hunk 2 above)
Comment (maczikasz) [reply src/test/java/com/gradle/enterprise/api/MockServer.java]: lol I have no idea why I used assert there :D probably pressed too early on the autocomplete

File: README.md [hunk 3]
```diff
@@ -15,6 +15,9 @@ $ ./gradlew install
 This builds and installs the program into `build/install/gradle-enterprise-api-samples`.
 You can use the `build/install/gradle-enterprise-api-samples/bin/gradle-enterprise-api-samples` script to run the sample.
 
+### Note on JDK11
+
+Due to a change in the [openapi generator project](https://github.com/gradle/gradle-enterprise-api-samples/pull/74)  JDK 11 is required to build the samples. The clients still work using java 8
```
Comment (BigDataWriter) [inline README.md:20]: Missing comma before 'JDK 11'.

Comment (BigDataWriter) [reply README.md:20]: Shouldn't that be 'JDK 8'?   Also, that sentence is missing a period.

**Review by sorin-florea — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (maczikasz) [reply README.md:20]: No it's not JDK 8, because running an app does not need a JDK but a JRE, but using JRE would've been kindof weird IMO

Comment (maczikasz) [inline README.md:20]:
```suggestion
Due to a change in the [openapi generator project](https://github.com/gradle/gradle-enterprise-api-samples/pull/74), JDK 11 is required to build the samples. The clients still work using java 8
```

File: build.gradle.kts [hunk 4]
```diff
@@ -32,18 +34,25 @@ dependencies {
...
+    testImplementation("org.junit.jupiter:junit-jupiter:5.7.1")
+    testRuntimeOnly("org.junit.platform:junit-platform-launcher")
+
 }
 
 val gradleEnterpriseVersion = "2023.2" // Must be later than 2022.1
 val baseApiUrl = providers.gradleProperty("apiManualUrl").orElse("https://docs.gradle.com/enterprise/api-manual/ref/")
 
 val apiSpecificationFileGradleProperty = providers.gradleProperty("apiSpecificationFile")
+val apiSpecificationURL = "${baseApiUrl.get()}gradle-enterprise-${gradleEnterpriseVersion}-api.yaml"
```
Comment (etiennestuder) [inline build.gradle.kts:48]: Calling get() defeats the purpose of the Provider API. Please rewrite the code so it remains Provider-API-based. If you get stuck, @wolfs can surely help.

File: build.gradle.kts [hunk 5]
```diff
@@ -73,6 +82,16 @@ openApiGenerate {
     ))
 }
 
+tasks.test {
+    useJUnitPlatform()
+
+    systemProperties["ge.api.url"] = apiSpecificationURL
+
+    java {
+        sourceCompatibility = JavaVersion.VERSION_1_8
```
Comment (etiennestuder) [inline build.gradle.kts:91]: Can't you use the toolchain API here, too?

**Review by etiennestuder — CHANGES_REQUESTED** [review]

**Review by etiennestuder — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (BigDataWriter) [reply README.md]: We should capitalize 'Java'.

**Pull request merged**